test/PositionToken.ts

```diff
@@ -1,115 +1,99 @@
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { expect } from "chai";
-import { ContractFactory } from "ethers";
-import { ethers } from "hardhat";
 import { PositionToken } from "../types/contracts";
-import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
+import { buildContractTestContext, ContractTestContext, signers } from "./ContractTestContext";
 
 describe("PositionToken test suit", function () {
     let r: ContractTestContext;
-    let ptContract: ContractFactory;
-
-    describe("Roles and initialization", async function () {
-        before(async () => {
-            r = await buildContractTestContext();
-            ptContract = await ethers.getContractFactory("PositionToken");
-        });
-
-        it("Should be built properly by ContractTestContext", async function () {
-            expect(r.positionToken).to.not.be.undefined;
-        });
-
-        it("Should not allow non admin to init", async function () {
-            const addr1PositionToken = await ptContract.deploy(r.addr1.address) as PositionToken;
-            const initPromise = addr1PositionToken.init(r.addr2.address);
-            await expect(initPromise).to.be.revertedWith("onlyAdmin: Caller is not admin");
+    const firstTokenId = 0;
+    const secondTokenId = 1;
+    const thirdTokenId = 2;
+    let firstTokenOwner: SignerWithAddress;
+    let firstTokenOwnerAddress: string;
+    let secondTokenOwner: SignerWithAddress;
+    let secondTokenOwnerAddress: string;
+    let thirdTokenOwner: SignerWithAddress;
+    let thirdTokenOwnerAddress: string;
+    let positionTokenAsExecutive: PositionToken;
+
+    before(async function () {
+        const [owner] = await signers;
+        r = await buildContractTestContext({
+            PositionToken: { executive: owner.address },
         });
```

Comment:
Roles are overridden per contract. One or more roles can be specified here to override. Normally in buildContractTestContext the PositionToken executive is LeverageEngine but in order to test as the current user we need to set it to owner.